### ISSUE TITLE
Android TV: Keyboard automatically open when pressing numeric key after Input usage

### ISSUE THREAD

**Issue by freeboub** [facebook/react-native#33559]
### Description

Keyboard automatically open when pressing numeric key (KEYCODE_1, ... ) after removing an InputText from the View (not tested on ios, but not sure it makes sens as there is no key number on apple TV)

### Version

0.66.3

### Output of `npx react-native info`

System:
    OS: Linux 5.16 Fedora Linux 35 (Workstation Edition)
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 4.73 GB / 15.14 GB
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 16.14.0 - /bin/node
    Yarn: 1.22.10 - /bin/yarn
    npm: 8.3.1 - /bin/npm
    Watchman: 0.0.0 - /bin/watchman
  SDKs:
    Android SDK:
      API Levels: 21, 24, 25, 28, 29, 30, 31, 32
      Build Tools: 20.0.0, 21.1.2, 24.0.3, 25.0.0, 25.0.1, 26.0.2, 28.0.3, 29.0.2, 29.0.3, 30.0.2, 31.0.0, 32.0.0
      System Images: android-29 | Android TV Intel x86 Atom, android-30 | Google TV Intel x86 Atom
      Android NDK: 20.1.5948944
  IDEs:
    Android Studio: Not Found
  Languages:
    Java: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: Not Found
    react-native-tvos:  0.66.3-3 
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Easy to reproduce with androidtv Device:
- Navigate in an app with an InputText
- close keyboard and leave the view
- press numerical key on remote

Sample project : https://github.com/freeboub/React-Native-tvos-demo/tree/sample_keyboard_issue

This is just a dummy InputText which is removed from main View onSubmit.
Once you end typing text, the view disappear.
At that point, if you press numerical key on remote control, the keyboard is automatically opened.
Keycode can be simulate with following adb command: 'adb shell input keyevent KEYCODE_1'

As a complementary test, if you initialize show state to false and input numerical Keycode, the issue is not reproduced.

Reproduced on multiple Androidtv devices and on emulator.

### Snack, code example, screenshot, or link to a repository

Sample basic project can be found here:
 https://github.com/freeboub/React-Native-tvos-demo/tree/sample_keyboard_issue

Issue I opened in react-native-tvos
https://github.com/react-native-tvos/react-native-tvos/issues/283

**Comment by freeboub** [comment]
I tried to debug the issue here are some additionnal informations:
- We use react-native-tvos, but it is reproduced with standard react-native version
- If I launch google voice assistant the incorrect behavior disappear.
- when the issue happen, android layout inspector confirm that the ReactEditText is well removed from the global layout
- When the issue present (keyboard opens when it should not), the android Activity doesn't receive onKeyDown ( but correctly receive onKeyUp, I add traces in ReactActivity.java )
- Override public boolean onKeyPreIme (int keyCode, KeyEvent event) in ReactEditText -> it is never called, I also try to implement it in ReactRootView, it is also not called. Notice That it is also not called in 'normal' use case.

**Comment by freeboub** [comment]
Here are some addtionnal informations:

to workaround the issue I tried to implement by own InputText component (I can share it if needed), but the same issue is reproduced with a custom component, so I think we can asume the issue is not directly linked to EditText management in react-native. It's really looks like a low level issue. Maybe a reference leak ...
I found an 'ugly' workaround which is working. When a TextInput is blured or unmounted, I launch an empty transparent activty. This activity just call finish onResume... Then the activity stack is correctly cleared and keyboard doesn't take the focus instead of our app. I causes a very small freeze in UI, but It is better than this issue.
